classes/migration/upgrade/OJSv3_3_0UpgradeMigration.inc.php

```diff
@@ -41,4 +43,121 @@ public function down() {
+			} else {
+				continue; // if parent class changes, the table is processed with other settings tables
+			}
+
+			$processedTables[] = $tableName;
+			foreach ($schema->properties as $propName => $propSchema) {
+				if (empty($propSchema->readOnly)) {
+					if ($propSchema->type === 'array' || $propSchema->type === 'object') {
+						Capsule::table($tableName)->where('setting_name', $propName)->get()->each(function ($row) use ($tableName) {
+							$this->_toJSON($row, $tableName, ['setting_name', 'locale'], 'setting_value');
+						});
+					}
+				}
+			}
+		}
+
+		// Convert settings where only setting_type column is available
+		$tables = Capsule::connection()->getDoctrineSchemaManager()->listTableNames();
+		foreach ($tables as $tableName) {
+			if (substr($tableName, -9) !== '_settings' || in_array($tableName, $processedTables)) continue;
+			Capsule::table($tableName)->where('setting_type', 'object')->get()->each(function ($row) use ($tableName) {
+				$this->_toJSON($row, $tableName, ['setting_name', 'locale'], 'setting_value');
+			});
+		}
+
+		// Finally, convert values of other tables dependent from DAO::convertToDB
+		Capsule::table('review_form_responses')->where('response_type', 'object')->get()->each(function ($row) {
+			$this->_toJSON($row, 'review_form_responses', ['review_id'], 'response_value');
+		});
+
+		Capsule::table('site')->get()->each(function ($row) {
+			$oldInstalledLocales = unserialize($row->{'installed_locales'});
+			$oldSupportedLocales = unserialize($row->{'supported_locales'});
+
+			if (is_array($oldInstalledLocales) && $this->_isNumerical($oldInstalledLocales)) $oldInstalledLocales = array_values($oldInstalledLocales);
+			if (is_array($oldSupportedLocales) && $this->_isNumerical($oldSupportedLocales)) $oldSupportedLocales = array_values($oldSupportedLocales);
+
+			if ($oldInstalledLocales) {
+				$newInstalledLocales = json_encode($oldInstalledLocales, JSON_UNESCAPED_UNICODE);
+				Capsule::table('site')->take(1)->update(['installed_locales' => $newInstalledLocales]);
+			}
+
+			if ($oldSupportedLocales) {
+				$newSupportedLocales = json_encode($oldSupportedLocales, JSON_UNESCAPED_UNICODE);
+				Capsule::table('site')->take(1)->update(['supported_locales' => $newSupportedLocales]);
+			}
+		});
+	}
+
+	/**
+	 * @param $row stdClass row representation
+	 * @param $tableName string name of a settings table
+	 * @param $searchBy array additional parameters to the where clause that should be combined with AND operator
+	 * @param $valueToConvert string column name for values to convert to JSON
+	 * @return void
+	 */
+	private function _toJSON($row, $tableName, $searchBy, $valueToConvert)
+	{
+		$oldValue = unserialize($row->{$valueToConvert});
+		// Reset arrays to avoid keys being mixed up
+		if (is_array($oldValue) && $this->_isNumerical($oldValue)) $oldValue = array_values($oldValue);
+		if (!$oldValue && !is_array($oldValue)) return; // don't continue if value cannot be unserialized
```

Comment:
Changed this line from `if (!$oldValue)` as empty arrays also can fall under this condition. I've noticed that in the test PostgreSQL dump in `publication_settings`, the value of `categoryIds` is empty serialized array. It will pass this check and `categoryIds` will be unset at the end.

Reply:
Hmm, interesting that the behavior with empty arrays is different in MySQL, at least on my test instance; probably related to Laravel's Query Builder.

Reply:
Hmm, yes, interesting -- maybe see if you can track this down as an expected/known behaviour? If not, and you can duplicate it with a small test script, maybe worth filing a new issue with Laravel. It would be interesting to see how responsive they are to cross-DB issues.

Reply:
Strange, and now the behavior is the same. Something else interacted with those empty arrays, not the current update script. Maybe the script had differences when I tested it with MySQL.